src/west/app/project.py

```diff
@@ -924,6 +924,29 @@ def fetch_strategy(self, args):
         else:
             return 'smart'
 
+    def update_submodules(self, project, submodules, call_location):
+        # Updates given project submodules by using
+        # 'git submodule update --init --checkout --recursive' command
+        # from the location passed through the call_location parameter value.
+        if submodules is not None:
+            if self.args.rebase:
+                submodules_update_strategy = '--rebase'
+            else:
+                submodules_update_strategy = '--checkout'
+            # For the list type, update given list of submodules.
+            if isinstance(submodules, list):
+                for submodule in submodules:
+                    if isinstance(submodule, Submodule):
+                        project.git(['-C', call_location, 'submodule',
+                                     'update', '--init',
+                                     submodules_update_strategy, '--recursive',
+                                     submodule.path])
+            # For the bool type, update all project submodules
+            elif isinstance(submodules, bool):
```

Comment:
https://www.google.com/search?q=isinstance+anti+pattern

You're making the internal implementation follow the user interface/configuration file exactly; it doesn't have to. You can have one `project.all_submodules` boolean on one hand and a possibly empty list of `project.submodules` as _another_ field. Combining them together in a Python `Union` unnecessarily mimics the user interface and makes IMHO the code more complicated.

Reply:
I think that this comment and the following concern the similar thing:
> BTW I think submodules: all/none would look better than submodules: true/false

I know that the code is a bit more complicated, however I wanted to keep it consistent with already existing fields in the manifest like `import`, for which we can assign boolean true/false values and a list. Isn't it better solution than adding new method of enabling/disabling the field in the manifest, which specific for only one case and different from the others?

Reply:
I missed the consistency aspect, makes sense. If @mbolivar-nordic is happy with this then I'm happy.